### PR TITLE
ODIM should poll the plugin and update the task status in DB

### PR DESCRIPTION
**Pull request by AngithCJ**

### CODE REVIEW

File: svc-task/thandle/tpolling.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+func (ts *TasksRPC) MonitorPluginTasks() {
+	duration := time.Duration(config.Data.PluginTasksConf.MonitorPluginTasksFrequencyInMins)
+	ticker := time.NewTicker(duration * time.Minute)
+	for range ticker.C {
+		ts.PollPlugin(context.TODO())
+	}
+}
+
+func (ts *TasksRPC) PollPlugin(ctx context.Context) {
+	l.LogWithFields(ctx).Info("Started polling plugin to monitor the plugin tasks")
```
Comment (hemkapoor) [inline svc-task/thandle/tpolling.go:56]: If we are having context.TODO() from calling function then process log will contain nil values.

File: svc-task/tmodel/plugin.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		l.LogWithFields(ctx).Errorf("failed to get taskmon response from %s(%s): %s: %+v",
+			plugin.ID, plugin.IP, err.Error(), response)
+		return nil, err
+	}
+	l.LogWithFields(ctx).Infof("Successfully got task response from %s(%s)", plugin.ID, plugin.IP)
+	return response, nil
+}
+
+// ContactPlugin is for sending requests to a plugin.
+func ContactPlugin(ctx context.Context, req PluginContactRequest, serverName string) (*http.Response, error) {
```
Comment (hemkapoor) [inline svc-task/tmodel/plugin.go]: This function is being used in other services aswell.. Can we move it to common lib and use.

File: lib-persistence-manager/persistencemgr/redis.go [hunk 3]
```diff
@@ -1007,6 +1007,53 @@ func (p *ConnPool) CreateTaskIndex(index string, value int64, key string) error
 	return nil
 }
 
+func (p *ConnPool) AddMemberToSet(key string, member string) *errors.Error {
```
Comment (amar-shalgar) [inline lib-persistence-manager/persistencemgr/redis.go:1017]: Please mention what these functions do

File: svc-task/thandle/tpolling.go (hunk 1 above)
Comment (AngithCJ) [reply svc-task/thandle/tpolling.go:56]: created context

File: svc-task/tmodel/plugin.go (hunk 2 above)
Comment (AngithCJ) [reply svc-task/tmodel/plugin.go]: moved to rest client library

**Review by hemkapoor — APPROVED** [review]

**Review by amar-shalgar — APPROVED** [review]

**Pull request merged**